Web Inspector: Remove the deprecated PageDebuggable

the-chenergy opened the pull request:
#### 7dcb5b690a092f621e586e0f28f066290d386f05
<pre>
Web Inspector: Remove the deprecated PageDebuggable
<a href="https://rdar.apple.com/164588069">rdar://164588069</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=302429">https://bugs.webkit.org/show_bug.cgi?id=302429</a>

Reviewed by BJ Burg.

Remove the deprecated page debuggable type since WebKitLegacy now
also creates a web-page debuggable as of 304467@main.

No new tests; observable inspection behavior should stay the same.
Inspector protocol generator and its tests are updated to stop using or
expecting the page debuggable.

* Source/JavaScriptCore/inspector/protocol/Animation.json:
* Source/JavaScriptCore/inspector/protocol/Audit.json:
* Source/JavaScriptCore/inspector/protocol/CPUProfiler.json:
* Source/JavaScriptCore/inspector/protocol/CSS.json:
* Source/JavaScriptCore/inspector/protocol/Canvas.json:
* Source/JavaScriptCore/inspector/protocol/Console.json:
* Source/JavaScriptCore/inspector/protocol/DOM.json:
* Source/JavaScriptCore/inspector/protocol/DOMDebugger.json:
* Source/JavaScriptCore/inspector/protocol/DOMStorage.json:
* Source/JavaScriptCore/inspector/protocol/Debugger.json:
* Source/JavaScriptCore/inspector/protocol/Heap.json:
* Source/JavaScriptCore/inspector/protocol/IndexedDB.json:
* Source/JavaScriptCore/inspector/protocol/Inspector.json:
* Source/JavaScriptCore/inspector/protocol/LayerTree.json:
* Source/JavaScriptCore/inspector/protocol/Memory.json:
* Source/JavaScriptCore/inspector/protocol/Network.json:
* Source/JavaScriptCore/inspector/protocol/Page.json:
* Source/JavaScriptCore/inspector/protocol/Recording.json:
* Source/JavaScriptCore/inspector/protocol/Runtime.json:
* Source/JavaScriptCore/inspector/protocol/ScriptProfiler.json:
* Source/JavaScriptCore/inspector/protocol/Security.json:
* Source/JavaScriptCore/inspector/protocol/Timeline.json:
* Source/JavaScriptCore/inspector/protocol/Worker.json:
* Source/JavaScriptCore/inspector/remote/RemoteControllableTarget.h:
* Source/JavaScriptCore/inspector/remote/RemoteInspectorConstants.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocket.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/scripts/codegen/models.py:
(validate_target_types):
* Source/JavaScriptCore/inspector/scripts/tests/command-targetType-matching-domain-debuggableType.json:
* Source/JavaScriptCore/inspector/scripts/tests/domain-targetType-matching-domain-debuggableType.json:
* Source/JavaScriptCore/inspector/scripts/tests/event-targetType-matching-domain-debuggableType.json:
* Source/JavaScriptCore/inspector/scripts/tests/expected/command-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/event-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-domain-debuggableTypes-value.json-error:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/PageInspectorController.cpp:
(WebCore::PageInspectorController::connectFrontend):
(WebCore::PageInspectorController::disconnectFrontend):
(WebCore::PageInspectorController::disconnectAllFrontends):
(WebCore::PageInspectorController::frontendInitialized):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::didChangeTitle):
(WebCore::FrameLoader::dispatchDidCommitLoad):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::~Page):
(WebCore::Page::inspectable const): Deleted.
(WebCore::Page::setInspectable): Deleted.
(WebCore::Page::remoteInspectionNameOverride const): Deleted.
(WebCore::Page::setRemoteInspectionNameOverride): Deleted.
(WebCore::Page::remoteInspectorInformationDidChange): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Page::inspectorDebuggable): Deleted.
(WebCore::Page::inspectorDebuggable const): Deleted.
* Source/WebCore/page/PageDebuggable.cpp: Removed.
* Source/WebCore/page/PageDebuggable.h: Removed.
* Source/WebCore/testing/Internals.cpp:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::setInspectable):
* Source/WebKit/Shared/DebuggableInfoData.cpp:
(WebKit::DebuggableInfoData::empty):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):

Canonical link: <a href="https://commits.webkit.org/305128@main">https://commits.webkit.org/305128@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5f0540cc561b2e1252eb92cdb5a155181a467769

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137581 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9942 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48869 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/145338 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90554 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8c9d77b0-1c58-4db0-958d-5014031d7e16) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/139453 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10645 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/10072 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/105228 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76824 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/00ce09f3-035a-4b2c-9995-831ec6653631) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140526 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7905 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/123301 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/86080 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ff850e56-352f-46f4-ae8d-885a5cd46b7f) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7539 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/5258 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5921 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/129542 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116911 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41459 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/148102 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/136102 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9624 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/42012 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113603 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9642 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/8112 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113946 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7463 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119541 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/64280 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21187 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9673 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37593 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/168851 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9404 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/73238 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/44056 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9613 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9465 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

burg commented on Source/JavaScriptCore/inspector/scripts/codegen/models.py:
```diff
@@ -47,13 +47,10 @@ def validate_target_types(debuggable_types, target_types):
             if 'service-worker' not in debuggable_types:
                 return False
-        elif target_type == 'web-page' or target_type == 'frame':
+        elif target_type in ('web-page', 'page', 'frame', 'worker'):
```
I don't think you meant to keep 'page' ?

the-chenergy replied:
This part is about the `page` target type, not debuggable type. (In combination with the [changes from a few lines above,](https://github.com/WebKit/WebKit/pull/55732/files/ad7d0383d6760e0f7d125195521918b7d72bc4bb#diff-4cc60fc6b69d7635c12c405696fec54083ac0438892d782a459ed89aed95b28bL50-L52) the `web-page` debuggable support is now expected to come with `page` and `worker` targets)

burg commented on Source/JavaScriptCore/inspector/scripts/codegen/models.py:
```diff
@@ -97,7 +94,7 @@ def validate_target_types(debuggable_types, target_types):
 }
 
-_ALLOWED_DEBUGGABLE_TYPE_STRINGS = ['itml', 'javascript', 'page', 'service-worker', 'web-page', 'wasm-debugger']
+_ALLOWED_DEBUGGABLE_TYPE_STRINGS = ['itml', 'javascript', 'service-worker', 'web-page', 'wasm-debugger']
```
Please ensure that any 'page' debuggable references in the code generator tests are fixed, and that the rebaselined test output is included in this diff.

burg reviewed (approved):
Wohoo!

the-chenergy committed:
Web Inspector: Remove the deprecated PageDebuggable
rdar://164588069
https://bugs.webkit.org/show_bug.cgi?id=302429

Reviewed by BJ Burg.

Remove the deprecated page debuggable type since WebKitLegacy now
also creates a web-page debuggable as of 304467@main.

No new tests; observable inspection behavior should stay the same.
Inspector protocol generator and its tests are updated to stop using or
expecting the page debuggable.

* Source/JavaScriptCore/inspector/protocol/Animation.json:
* Source/JavaScriptCore/inspector/protocol/Audit.json:
* Source/JavaScriptCore/inspector/protocol/CPUProfiler.json:
* Source/JavaScriptCore/inspector/protocol/CSS.json:
* Source/JavaScriptCore/inspector/protocol/Canvas.json:
* Source/JavaScriptCore/inspector/protocol/Console.json:
* Source/JavaScriptCore/inspector/protocol/DOM.json:
* Source/JavaScriptCore/inspector/protocol/DOMDebugger.json:
* Source/JavaScriptCore/inspector/protocol/DOMStorage.json:
* Source/JavaScriptCore/inspector/protocol/Debugger.json:
* Source/JavaScriptCore/inspector/protocol/Heap.json:
* Source/JavaScriptCore/inspector/protocol/IndexedDB.json:
* Source/JavaScriptCore/inspector/protocol/Inspector.json:
* Source/JavaScriptCore/inspector/protocol/LayerTree.json:
* Source/JavaScriptCore/inspector/protocol/Memory.json:
* Source/JavaScriptCore/inspector/protocol/Network.json:
* Source/JavaScriptCore/inspector/protocol/Page.json:
* Source/JavaScriptCore/inspector/protocol/Recording.json:
* Source/JavaScriptCore/inspector/protocol/Runtime.json:
* Source/JavaScriptCore/inspector/protocol/ScriptProfiler.json:
* Source/JavaScriptCore/inspector/protocol/Security.json:
* Source/JavaScriptCore/inspector/protocol/Timeline.json:
* Source/JavaScriptCore/inspector/protocol/Worker.json:
* Source/JavaScriptCore/inspector/remote/RemoteControllableTarget.h:
* Source/JavaScriptCore/inspector/remote/RemoteInspectorConstants.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorSocket.cpp:
(Inspector::RemoteInspector::listingForInspectionTarget const):
* Source/JavaScriptCore/inspector/scripts/codegen/models.py:
(validate_target_types):
* Source/JavaScriptCore/inspector/scripts/tests/command-targetType-matching-domain-debuggableType.json:
* Source/JavaScriptCore/inspector/scripts/tests/domain-targetType-matching-domain-debuggableType.json:
* Source/JavaScriptCore/inspector/scripts/tests/event-targetType-matching-domain-debuggableType.json:
* Source/JavaScriptCore/inspector/scripts/tests/expected/command-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/domain-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/event-targetType-matching-domain-debuggableType.json-result:
* Source/JavaScriptCore/inspector/scripts/tests/expected/fail-on-domain-debuggableTypes-value.json-error:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/PageInspectorController.cpp:
(WebCore::PageInspectorController::connectFrontend):
(WebCore::PageInspectorController::disconnectFrontend):
(WebCore::PageInspectorController::disconnectAllFrontends):
(WebCore::PageInspectorController::frontendInitialized):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::didChangeTitle):
(WebCore::FrameLoader::dispatchDidCommitLoad):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::~Page):
(WebCore::Page::inspectable const): Deleted.
(WebCore::Page::setInspectable): Deleted.
(WebCore::Page::remoteInspectionNameOverride const): Deleted.
(WebCore::Page::setRemoteInspectionNameOverride): Deleted.
(WebCore::Page::remoteInspectorInformationDidChange): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Page::inspectorDebuggable): Deleted.
(WebCore::Page::inspectorDebuggable const): Deleted.
* Source/WebCore/page/PageDebuggable.cpp: Removed.
* Source/WebCore/page/PageDebuggable.h: Removed.
* Source/WebCore/testing/Internals.cpp:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
(WebCore::ServiceWorkerThreadProxy::setInspectable):
* Source/WebKit/Shared/DebuggableInfoData.cpp:
(WebKit::DebuggableInfoData::empty):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):

Canonical link: https://commits.webkit.org/305128@main
